Foundation/NSEnergyFormatter.swift

```diff
@@ -7,41 +7,195 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
 extension EnergyFormatter {
-    public enum Unit : Int {
+    public enum Unit: Int {
+
+        case joule = 11
+        case kilojoule = 14
+        case calorie = 1793 // chemistry "calories", abbr "cal"
+        case kilocalorie = 1794 // kilocalories in general, abbr “kcal”, or “C” in some locales (e.g. US) when usesFoodEnergy is set to YES
+
+        // Map Unit to UnitEnergy class to aid with conversions
+        fileprivate var unitEnergy: UnitEnergy {
+            switch self {
+            case .joule:
+                return UnitEnergy.joules
+            case .kilojoule:
+                return UnitEnergy.kilojoules
+            case .calorie:
+                return UnitEnergy.calories
+            case .kilocalorie:
+                return UnitEnergy.kilocalories
+            }
+        }
+
+        // Reuse symbols defined in UnitEnergy, except for kilocalories, which is defined as "kCal"
+        fileprivate var symbol: String {
+            switch self {
+            case .kilocalorie:
+                return "kcal"
+            default:
+                return unitEnergy.symbol
+            }
+        }
+
+        // Return singular, full string representation of the energy unit
+        fileprivate var singularString: String {
+            switch self {
+            case .joule:
+                return "joule"
```

Comment:
Certainly. In the meantime, however, is it appropriate for a formatter that is unable to localize for some particular locales and unit styles offer known incorrect output when asked, or should the code check for supported locales and otherwise bail with `NSUnimplemented()`?